src/hazelcore/UHExp.re

```diff
@@ -305,10 +305,75 @@ let text_operand =
     );
   };
 
-let associate = (seq: seq) => {
-  let skel_str = Skel.mk_skel_str(seq, Operators_Exp.to_parse_string);
-  let lexbuf = Lexing.from_string(skel_str);
-  SkelExprParser.skel_expr(SkelExprLexer.read, lexbuf);
+let associate = (seq: seq): skel => {
```

Comment:
can you leave a comment pointing to the reference material that you used for implementing this? hard to understand what is going on here without that...

Reply:
Yes I'll leave a ref to the wikipedia page, and add some comments explaining what each component does.

Reply:
@cyrus- Take a second look